src/rf/apps/workers/process.py

```diff
@@ -265,12 +276,13 @@ def emr_heartbeat(self, record):
 
         if not check_cluster_status(job_id):
             log.info('EMR job for layer %d has failed!', layer_id)
-            return status_updates.update_layer_status(layer_id,
-                                                      enums.STATUS_FAILED,
-                                                      ERROR_MESSAGE_EMR_DEAD)
+            return status_updates.mark_layer_status_end(layer_id,
+                                                        enums.STATUS_FAILED,
+                                                        ERROR_MESSAGE_EMR_DEAD)
```

Comment:
Not necessarily. We could have gotten pretty far, including initially creating a cluster but then failed later on.